### PR TITLE
feat: improvising Aggregated monitoring dashboard

### PR DESCRIPTION
**Pull request by asthamohta**
This PR does the following thing:
1) Adds job id to the title of the aggregated dashboard
2) adds a dashboard at the bottom with shard to shard dashboard mapping
3) adds a tile in the dataflow section which shows cpu utilisation for each shard
4) Some general cleanup of the code

### CODE REVIEW

File: streaming/streaming.go [hunk 1]
```diff
@@ -533,11 +533,14 @@ func CleanUpStreamingJobs(ctx context.Context, conv *internal.Conv, projectID, r
...
+	if conv.Audit.StreamingStats.MonitoringResources.DashboardName != "" && !conv.IsSharded {
+		CleanupMonitoringDashboard(ctx, conv.Audit.StreamingStats.MonitoringResources.DashboardName, projectID)
 	}
-	if conv.Audit.StreamingStats.AggMonitoringDashboard.DashboardName != "" && conv.IsSharded {
-		CleanupMonitoringDashboard(ctx, conv.Audit.StreamingStats.AggMonitoringDashboard.DashboardName, projectID)
+	if conv.Audit.StreamingStats.AggMonitoringResources.DashboardName != "" && conv.IsSharded {
+		CleanupMonitoringDashboard(ctx, conv.Audit.StreamingStats.AggMonitoringResources.DashboardName, projectID)
+	}
+	if conv.Audit.StreamingStats.AggMonitoringResources.DashboardName != "" && conv.IsSharded {
+		CleanupMonitoringDashboard(ctx, conv.Audit.StreamingStats.AggMonitoringResources.DashboardName, projectID)
```
Comment (manitgupta) [inline streaming/streaming.go:543]: Is this a rebase error? Seems identical.

File: common/metrics/queries.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ const (
 		"filter (resource.instance_id == '%s') | group_by 1m, [value_utilization_mean: mean(value.utilization)] " +
 		"| every 1m | group_by [], [value_utilization_mean_aggregate: aggregate(value_utilization_mean)]"
 	spannerStorageUtilDbQuery = "fetch spanner_instance | metric 'spanner.googleapis.com/instance/storage/used_bytes' | " +
-		"filter (resource.instance_id == '%s') && (metric.database == '%s') | " +
+		"filter (metric.database == '%s') && (resource.instance_id == '%s') | " +
```
Comment (bharadwaj-aditya) [inline common/metrics/queries.go:68]: This this swap intentional ?

**Review by bharadwaj-aditya — APPROVED** [review]
Only 1 query. LGTM Overall

It would be great if you could check and confirm on the retrial/error handling in case of failure of creation of these dashboards. I was not able to see that handling in this PR

File: common/metrics/queries.go (hunk 2 above)
Comment (asthamohta) [reply common/metrics/queries.go:68]: Yes, remember I mentioned there a bug, the issue was that I put the instance name in db name and vice versa 😅

File: streaming/streaming.go (hunk 1 above)
Comment (asthamohta) [reply streaming/streaming.go:543]: Yes, sorry it got missed

File: common/metrics/queries.go (hunk 2 above)
Comment (asthamohta) [reply common/metrics/queries.go:68]: Also for error handline, that is part of this PR: https://github.com/GoogleCloudPlatform/spanner-migration-tool/pull/686
I have tested failure scenarios as mentioned in the description

**Review by manitgupta — APPROVED** [review]

**Pull request merged**